### ISSUE TITLE
Master branch not pre-building

### ISSUE THREAD

**Issue by dunnry** [gitpod-io/gitpod#732]
**Describe the bug**
We are not seeing our master images getting pre-built.

**To Reproduce**
Make a commit to master, wait a few hours and launch gitpod on master branch.

**Expected behavior**
We were expecting to see most of the yarn build process already done when the dev environment launched.

.gitpod.yml 
```
ports:
- port: 3000
tasks:
  - init: >
      language-server/gradlew -p language-server assemble &&
      yarn --cwd theia
    command: >
      yarn --cwd theia/browser-app start --hostname 0.0.0.0
github:
  prebuilds:
    # enable for the master/default branch (defaults to true)
    master: true
    # enable for all branches in this repo (defaults to false)
    branches: true
    # enable for pull requests coming from this repo (defaults to true)
    pullRequests: true
    # enable for pull requests coming from forks (defaults to false)
    pullRequestsFromForks: true
    # add a "Review in Gitpod" button as a comment to pull requests (defaults to true)
    addComment: true
    # add a "Review in Gitpod" button to pull requests (defaults to false)
    addBadge: false
    # add a label once the prebuild is ready to pull requests (defaults to false)
    addLabel: prebuilt-in-gitpod
```

**Comment by svenefftinge** [comment]
Did you install the [GitHub App](https://www.gitpod.io/docs/46_prebuilds/) on your repository?

If that's the case, it might be that during the outage (we had two hours ago in the US cluster) a prebuild was canceled. You can manually trigger a prebuild using the following context URL:

https://gitpod.io/#prebuild/<repo-URL>

**Comment by dunnry** [comment]
I will check with the repository owner.  @JanKoehnlein I didn't realize this was a setting on an app in the repository, but that kinda makes sense now.  Can you check?  This used to be working.

**Comment by JanKoehnlein** [comment]
The app is installed and the context URL yields an error 
```
Cannot (re-)start a headless workspace.
Please file an issue if you think this is a bug.
```

**Comment by svenefftinge** [comment]
That error is normal. The manual prefix URL is not meant to be used broadly :-) 
If you start a workspace afterward you'll see it has a prebuild.

**Comment by dunnry** [comment]
I am still not seeing any pre-builds.  Is this something we can get fixed?

**Comment by jankeromnes** [comment]
I'm also having a prebuild problem in #753. I'll check if this issue is the same problem.

**Comment by jankeromnes** [comment]
Update: It's not the same problem as #753, as prebuilds for your repo seem to work well (in #753 they time out).

However, they indeed don't seem to trigger on `master` pushes anymore. Since April, prebuilds have become so rare that I assume they're all triggered manually now.

Here are the last 10 prebuilds (all successful):

```
+----------------------------+
| creationTime               |
+----------------------------+
| 2019-08-20 14:48:59.504225 |
| 2019-08-14 16:42:59.820006 |
| 2019-06-12 13:23:58.464459 |
| 2019-06-12 12:34:07.290541 |
| 2019-06-12 12:09:56.421421 |
| 2019-05-22 15:27:21.117191 |
| 2019-05-01 18:02:00.791584 |
| 2019-05-01 16:12:08.753959 |
| 2019-04-17 15:29:03.379968 |
| 2019-04-15 09:44:15.233330 |
+----------------------------+
```

**Comment by JanKoehnlein** [comment]
We also had some PR branches in the meantime. Haven't they been pre-built either?

**Comment by jankeromnes** [comment]
> We also had some PR branches in the meantime. Haven't they been pre-built either?

Unfortunately, no. The prebuilds I listed are for all branches.

**Comment by jankeromnes** [comment]
@svenefftinge noticed an error message from the Gitpod App, and re-installed it. That seems to have fixed the triggering of prebuilds:

```
+-----------+-------------+----------------------------+
| state     | error       | creationTime               |
+-----------+-------------+----------------------------+
| available |             | 2019-08-30 08:21:11.283919 |
| available | task failed | 2019-08-29 17:50:17.913703 |
| available | task failed | 2019-08-29 15:44:34.097890 |
| available |             | 2019-08-29 13:05:24.940592 |
| available |             | 2019-08-20 14:48:59.504225 |
| available |             | 2019-08-14 16:42:59.820006 |
| available |             | 2019-06-12 13:23:58.464459 |
| available |             | 2019-06-12 12:34:07.290541 |
| available |             | 2019-06-12 12:09:56.421421 |
| available |             | 2019-05-22 15:27:21.117191 |
+-----------+-------------+----------------------------+
```

(The `task failed` message is probably due to temporary 503 errors I noticed on https://oss.sonatype.org/ yesterday.)

We need to get better at noticing/handling these kinds of errors, but other than that, I think the issue is now fixed! 🎉

**Comment by dunnry** [comment]
This is still happening.  I don't see any PR build getting pre-built.

**Comment by JanKoehnlein** [comment]
I see pre-builds as expected. 

Only when I click the Gitpod button in the comments of a PR before the Gitpod app adds the check, I get a new self-building WS. As soon as the check is there, I see the pre-build logs when clicking it. @dunnry Could this have been your problem?

**Comment by dunnry** [comment]
I will keep an eye on it.  I had reviewed a PR that was several hours old and wasn't pre-built.
